packages/fast-form-generator-react/src/form/form.utilities.ts

```diff
@@ -218,6 +227,156 @@ export function getComponentTracker(
+        }
+
+        reconstitutedDataLocation += reconstitutedDataLocation === "" ? dataLocationStrings[i] : `.${dataLocationStrings[i]}`;
+    }
+
+    return schemaLocationStrings;
+}
+
+/**
+ * Gets data from a data and location
+ */
+export function getPartialData(location: string, data: any): any {
+    return location === "" ? data : get(data, location);
+}
+
+/**
+ * Gets an array of oneOf/anyOf with a valid index from a schema and data
+ */
+export function getSchemaOneOfAnyOfLocationStrings(schema: any, data: any): string[] {
+    const schemaLocationStrings: string[] = [];
+
+    if (!!schema.anyOf) {
+        schemaLocationStrings.push(`anyOf.${getValidAnyOfOneOfIndex("anyOf", data, schema)}`);
+    }
+
+    if (!!schema.oneOf) {
+        schemaLocationStrings.push(`oneOf.${getValidAnyOfOneOfIndex("oneOf", data, schema)}`);
+    }
+
+    return schemaLocationStrings;
+}
+
+/**
+ * Get an array of schema location strings from a single data location item
+ */
+export function getSchemaLocationStringsFromDataLocationString(config: ISchemaLocationStringsFromDataLocationStringConfig): string[] {
+    const schemaLocationStrings: string[] = [];
+
+    if (config.isObject && !config.isArray) {
+        schemaLocationStrings.push("properties");
+    }
+
+    schemaLocationStrings.push(config.dataLocation);
+
+    if ((Array.isArray(config.data) || checkDataLocationIsArrayItem(config.dataLocation)) && config.endOfContainingString) {
+        schemaLocationStrings.push("items");
+    }
+
+    return schemaLocationStrings;
+}
+
+/**
+ * Checks to see if the data location item is an array item
+ */
+export function checkDataLocationIsArrayItem(dataLocationItem: string): boolean {
+    const squareBracketRegex: RegExp = /\[(\d+?)\]/g;
+    const match: boolean = false;
+
+    if (dataLocationItem.match(squareBracketRegex)) {
+        const matches: string[] = dataLocationItem.match(squareBracketRegex);
+
+        if (typeof parseInt(matches[0].replace(squareBracketRegex, "$1"), 10) === "number") {
+            return true;
+        }
+    }
+
+    return match;
+}
+
+/**
+ * Converts a data location strings array items into bracket notation
+ */
+export function convertArrayItemsToBracketNotation(dataLocation: string, data: any): string {
+    const normalizedDataLocation: string[] = [];
+    const dataLocations: string[] = dataLocation.split(".");
+
+    for (let i: number = 0; i < dataLocations.length; i++) {
+        if (Array.isArray(get(normalizedDataLocation, data))) {
+            normalizedDataLocation.push(`${dataLocations[i]}[${dataLocations[i + 1]}]`);
+            i++;
+        } else {
+            normalizedDataLocation.push(dataLocations[i]);
+        }
+    }
+
+    return normalizedDataLocation.join(".");
+}
+
+/**
+ * Gets the index from a JSON schemas oneOf/anyOf array that validates against the data
+ */
+export function getValidAnyOfOneOfIndex(oneOfAnyOf: string, data: any, schema: any): number {
+    let validIndex: number;
```

Comment:
You should use `findIndex` - it will break out of the loop once the index is found:

```
return schema[oneofAnyOf].findIndex((item) => tv4.validate(data, item);
```

You'll also need to validate for a `-1` return value whereever this is called.